bin/generate_npm_tarball.sh

```diff
@@ -6,13 +6,25 @@ if [ $# -ne 2 ]; then
 fi
 package=$1
 output=$2
-
 wd=$(mktemp -d)
 trap "rm -r '$wd'" EXIT INT TERM
 
-mkdir $wd/cache $wd/install
+create_cache() {
+  node_modules/npm/bin/npm-cli.js install --cache $wd/cache $package --production --verbose
+}
+
+mkdir $wd/cache-combined $wd/cache $wd/install
 cd $wd/install
-npm install --cache $wd/cache $package --production --verbose
+npm install npm@1.3.x
+create_cache
+# save el7 (which uses 1.3.6 npm) results before running it again
+find $wd/cache -name .cache.json -exec cp -iv --parents -t $wd/cache-combined/ {} +
```

Comment:
`find $wd/cache .... cp --parents` is creating parent paths for the entire directory path, e.g. `tmp/tmp.v298lMRyn3/cache/registry.npmjs.org/yargs` in the resulting cache tarball. You could perhaps do `(cd $wd/cache && find . -name ...)`?

(Also consider removing `-i` for interactive mode, just in case)